Update/product details

camposjoaoc opened the pull request:
**ProductDetails: Dynamic tasks from backend + Zustand cart update**

Description:

Now fetching and showing plant tasks dynamically from `/plants/:id/tasks`
Tasks are styled with Tailwind for better visibility
Compatible with the new Zustand cart actions and QuantityControl
Improved loading and not-found states for a smoother UX

Let me know if you think anything should be changed or improved!

reevcrona reviewed (approved):
LGTM! Great job!

roztownsend commented on frontend/vite-project/src/pages/ProductDetails.tsx:
```diff
@@ -12,6 +12,7 @@ const ProductDetails: React.FC = () => {
   const [plant, setPlant] = useState<ProductItem | null>(null);
   const [loading, setLoading] = useState(true);
   const { addItem } = useCartActions();
+  const [tasks, setTasks] = useState<{ id: string; description: string }[]>([]);
```
```suggestion
  const [tasks, setTasks] = useState<{ id: string; description: string; start_month: number; end_month: number }[]>([]);
```

roztownsend commented on frontend/vite-project/src/pages/ProductDetails.tsx:
```diff
@@ -32,6 +33,20 @@ const ProductDetails: React.FC = () => {
+    };
+    fetchTasks();
+  }, [id]);
+
```
There is probably a better way to do this, but 

```suggestion
  const months = ["", "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
```

roztownsend commented on frontend/vite-project/src/pages/ProductDetails.tsx:
```diff
@@ -84,18 +99,17 @@ const ProductDetails: React.FC = () => {
+                tasks.map((task) => (
+                  <li key={task.id} className="product-task-item">
+                    <span role="img" aria-label="task">🌱</span>
+                    <strong>Task:</strong> {task.description}
```
```suggestion
                    <strong>{months[task.start_month]} to {months[task.end_month]}</strong> {task.description}
                  </li>
```

You'll need to add some additional logic here in cases where the task starts and ends on the same month. I would have it default to the zero index which is a blank string, but again, there may be better ways to do it.

roztownsend reviewed (changes requested):
Looks good. It's more of a nice-to-have, but showing a variation of the product name (

camposjoaoc commented on backend/src/routes/plantRoutes.ts:
```diff
@@ -73,4 +74,16 @@ router.get("/search", async (req: Request, res: Response): Promise<void> => {
+// GET /plants/:id/tasks
+router.get("/:id/tasks", async (req: Request, res: Response) => {
+  try {
+    const plantId = req.params.id;
+    const tasks = await Task.findAll({ where: { plant_id: plantId } });
+    res.json(tasks);
```
Thanks for the suggestion! That could work — I’ll give it a try and see how it goes. And yeah, I’ll add the method to Plant.model.ts 🚀

camposjoaoc commented:
Made adjustments based on feedback from Jacob and Ros:

Improve product details and tasks display
	•	Grouped and displayed plant tasks by description and month range on the product details page
	•	Updated routes and product card to use friendly URLs with product name slugs
	•	Added and used a shared Task type for better type safety
	•	Applied small improvements to task fetching and rendering logic

roztownsend reviewed (approved):
Nice 👍

camposjoaoc commented:
About the product name in the URL @roztownsend — I’m using the slugify function in the ProductCard component. It just turns the product name into a URL-friendly format by making it lowercase, replacing spaces with dashes, and removing special characters.

Since the product page fetches data based on the product ID (which I’m already passing in the URL), I decided to keep the ID there and just add the product name at the end.

Changing it to fetch by name would mean rewriting that whole logic, which doesn’t feel necessary right now — especially since everything’s working fine. And honestly, using the ID feels more reliable anyway.
